File: src/GasStation.Domain/Entities/Invoice.cs
using GasStation.Domain.Enums;
    
namespace GasStation.Domain.Entities;

public class Invoice
{
    public int Id { get; set; }
    public string Title { get; set; } = null!;
    public long CreatedDate { get; set; }
    public TransactionType TransactionType { get; set; }
    public string Consumer { get; set; } = null!;
    public string Provider { get; set; } = null!;        
    public decimal TotalPrice { get; set; } 
    public double TotalFuelQuantity { get; set; }
    public Fuel Fuel { get; set; } = null!;
    public IEnumerable<Report> Reports { get; set; } = null!;
}      

File: src/GasStation.UI/src/app/models/invoice/Invoice.ts
import {Fuel} from "../fuel/Fuel";

export class Invoice{
  id?: number;
  title?: string;
  createdDate?: string;
  transactionType?: string;
  consumer?: string;
  provider?: string;
  totalPrice?: number;
  totalFuelQuantity?: number;
  fuel?: Fuel;
}
